homeassistant/components/systemmonitor/coordinator.py

```diff
@@ -0,0 +1,144 @@
+"""DataUpdateCoordinators for the System monitor integration."""
+from __future__ import annotations
+
+from abc import abstractmethod
+from datetime import datetime
+import logging
+import os
+from typing import TypeVar
+
+import psutil
+from psutil._common import sdiskusage, shwtemp, snetio, snicaddr, sswap
+from psutil._pslinux import svmem
```

Comment:
This will fail on macos. Can you guard with if TYPE_CHECKING?

Reply:
Thanks for reminding me on that.
We need to import the right `svmem` depending on platform like `psutil`does or perhaps even easier construct our own instead.

Reply:
E   ImportError: cannot import name '_psutil_linux' from 'psutil' (/Users/bdraco/home-assistant/venv/lib/python3.12/site-packages/psutil/__init__.py). Did you mean: '_psutil_osx'?

Reply:
Yes. Didn't have time to fix that before my flight but I'll do that tomorrow.

As we anyhow only use a certain portion I'll just create my own data class so it's platform independent.

Reply:
Kinda annoying that they don't have it available for typing at top level. Looks like its never imported up
```
psutil/_psaix.py:svmem = namedtuple('svmem', ['total', 'available', 'percent', 'used', 'free'])
psutil/_psaix.py:    return svmem(total, avail, percent, inuse, free)
psutil/_psbsd.py:svmem = namedtuple(
psutil/_psbsd.py:    'svmem', ['total', 'available', 'percent', 'used', 'free',
psutil/_psbsd.py:    return svmem(
psutil/_pslinux.py:svmem = namedtuple(
psutil/_pslinux.py:    'svmem', ['total', 'available', 'percent', 'used', 'free',
psutil/_pslinux.py:    return svmem(
psutil/_psosx.py:svmem = namedtuple(
psutil/_psosx.py:    'svmem', ['total', 'available', 'percent', 'used', 'free',
psutil/_psosx.py:    return svmem(total, avail, percent, used, free, active, inactive, wired)
psutil/_pssunos.py:svmem = namedtuple('svmem', ['total', 'available', 'percent', 'used', 'free'])
psutil/_pssunos.py:    return svmem(total, avail, percent, used, free)
psutil/_pswindows.py:svmem = namedtuple('svmem', ['total', 'available', 'percent', 'used', 'free'])
psutil/_pswindows.py:    return svmem(total, avail, percent, used, free)
```

Reply:
Also there are 3 more fields for BSD `active, inactive, wired`

Reply:
We're not using them so therefore I think it's just easier to make our own class here. 

Don't really understand why they don't have this top-level but it's like that so we just have to work around it.

Reply:
I think with the last changes this should work also for your tests so would appreciate a go.
Then I think it's ready for review

Reply:
```
tests pass on mac os
 tests/components/systemmonitor/test_config_flow.py ✓✓✓✓✓                                                                                                                               24% ██▍       
 tests/components/systemmonitor/test_init.py ✓✓                                                                                                                                         33% ███▍      
 tests/components/systemmonitor/test_sensor.py ✓✓✓✓✓✓✓✓✓✓✓                                                                                                                              86% ████████▋ 
 tests/components/systemmonitor/test_util.py ✓✓✓                                                                                                                                       100% ██████████
-------------------------------------------------------------------------------------- snapshot report summary ---------------------------------------------------------------------------------------
74 snapshots passed.

Results (1.63s):
      21 passed
```